### ISSUE TITLE
Pre-trained VGG16 weights & ResNet18, 34 support

### ISSUE THREAD

**Issue by JuheonYi** [Timforce/tiny-tf#9]
Hi, sorry to bother you again.
May I ask if you could provide me the pre-trained VGG16 caffe model or VGG16.npy file, in case you have them?

I have successfully converted ResNet50.npy with the caffe-to-tensorflow tool you told me using the official ResNet50.caffemodel at https://github.com/KaimingHe/deep-residual-networks, 
but I could not find the right one for VGG16 :(

In case you have 'em, it would be great if you could share it!

And one more, do you plan on adding ResNet-18 and 34 to the codes?
I am currently trying to add them as well, but I wondered if you already have them, or plan to add them.

Thanks a lot!

**Comment by Timforce** [comment]
Hi,
Here's the [source][VGGnet] for both VGG16 and VGG19 caffe models.
And sorry that i do not have ResNet-18 and 34 version, nor do i plan to add them at this moment.
I'm more interested in improving network training efficiency and stability, since there still exist a small gap of both between this repo and [MatConvNet version][tiny-repo].

[VGGnet]: http://www.robots.ox.ac.uk/~vgg/research/very_deep/
[tiny-repo]: https://github.com/peiyunh/tiny/

**Comment by JuheonYi** [comment]
Hi, thanks for the info! 
I have successfully managed to convert VGG16 model, after some modifications in the prototxt file for recent caffe versions, as indicated in https://github.com/ethereon/caffe-tensorflow/issues/39

In case I am done with ResNet-18 and 34, I will share it :)
Glad to hear that you are working on network training efficiency. Hope to see the updates soon!

Thanks for all the help.

**Comment by JuheonYi** [comment]
Oh, and one more, when I try to load Resnet50.npy, I am getting a bunch of "Ignore ..." messages.
As I am not getting such messages except for the res5xx blocks when loading the Resnet101.npy file you provided, seems that I made some mistakes in converting the caffemodel :(

I recall that you said you made some modifications when using the caffe to tensorflow tool.
Did it include some additional processes in the converting process, such as changing the weight names or so on?

this is the .npy file I converted myself, by the way.
https://drive.google.com/file/d/1LZ6U8aHFz7CP5Fx21IAZAqz72k-EBcu0/view?usp=sharing

original resnet 50 caffe model I used:
https://onedrive.live.com/?authkey=%21AAFW2-FVoxeVRck&id=4006CBB8476FF777%2117887&cid=4006CBB8476FF777

Thanks,
Juheon

**Comment by JuheonYi** [comment]
Oh, and one more, when I try to load Resnet50.npy I converted myself, I am getting a bunch of "Ignore ..." messages with regarding bach normalization layers such as below:
(I printed the ValueError in load() function in lib/networks/network.py)
ignore bn2b_branch2a value error:Variable bn2b_branch2a/mean does not exist, or was not created with tf.get_variable(). Did you mean to set reuse=tf.AUTO_REUSE in VarScope?

Seems assigning weights to conv works fine, but my problem seems to lie in assigning weights for batch normalization layers.

As I am not getting such messages except for the res5xx blocks when loading the Resnet101.npy file you provided, seems that I made some mistakes in converting the caffemodel :(

I recall that you said you made some modifications when using the caffe to tensorflow tool.
Did it include some additional processes in the converting process, such as changing the weight names or so on?

this is the Resnet50.npy file I converted myself, by the way.
https://drive.google.com/file/d/1LZ6U8aHFz7CP5Fx21IAZAqz72k-EBcu0/view?usp=sharing

original resnet 50 caffe model I used:
https://onedrive.live.com/?authkey=%21AAFW2-FVoxeVRck&id=4006CBB8476FF777%2117887&cid=4006CBB8476FF777

Thanks,
Juheon

**Comment by Timforce** [comment]
Hi,
I did make some tweaks to weight names, as you can see in down below:
![snap2](https://user-images.githubusercontent.com/13286580/56906446-d1fadb00-6ad4-11e9-9671-432fd00632a4.png)

This is because i use [contrib version(with more customizable arguments)][tf_contrib_batch_norm] of batch normalization rather than [nn version][tf_nn_batch_norm], and they have different naming pattern for weights. 
You can add naming modification in the [convert.py][caffe2TF_network].
```python
for var in data:
    if var[:2] == 'bn':
        data[var]['moving_mean'] = data[var].pop('mean')
        data[var]['moving_variance'] = data[var].pop('variance')
        data[var]['gamma'] = data[var].pop('scale')
        data[var]['beta'] = data[var].pop('offset')
```

[tf_contrib_batch_norm]: https://www.tensorflow.org/api_docs/python/tf/contrib/layers/batch_norm
[tf_nn_batch_norm]: https://www.tensorflow.org/api_docs/python/tf/nn/batch_normalization
[caffe2TF_network]: https://github.com/ethereon/caffe-tensorflow/blob/master/convert.py#L30

**Comment by JuheonYi** [comment]
Hi, thanks for the answer.
Always getting great help from you :)
I will try it out!

**Comment by MansiAgarwal11** [comment]
hello @JuheonYi 
what all changes did you make to run the script with VGG16 and Resnet50 backbone?
In addition, could you send me the links to the .npy files?
